source/ota.c

```diff
@@ -2811,54 +2820,64 @@ static uint32_t searchTransition( const OtaEventMsg_t * pEventMsg )
     return i;
 }
 
-/* Event Processing loop to run the OTA state machine. */
-void OTA_EventProcessingTask( void * pUnused )
+static void receiveAndProcessOtaEvent( void )
 {
     OtaEventMsg_t eventMsg = { 0 };
     uint32_t i = 0;
     uint32_t transitionTableLen = ( uint32_t ) ( sizeof( otaTransitionTable ) / sizeof( otaTransitionTable[ 0 ] ) );
 
-    ( void ) pUnused;
+    if( otaAgent.pOtaInterface == NULL )
+    {
+        LogError( ( "Failed to receive event: OS Interface not set" ) );
+    }
 
     /*
-     * OTA Agent is ready to receive and process events so update the state to ready.
+     * Receive the next event form the OTA event queue to process.
      */
-    otaAgent.state = OtaAgentStateReady;
-
-    while( otaAgent.state != OtaAgentStateStopped )
+    if( otaAgent.pOtaInterface->os.event.recv( NULL, &eventMsg, 0 ) == OtaOsSuccess )
```

Comment:
Isn't it a null pointer access if `otaAgent.pOtaInterface == NULL`?

Reply:
Yes it will be, currently the code is making an assumption that these interfaces are being set before the library is initialized. This needs to be fixed in multiple places to be resolved. I'll be adding unit tests to cover these cases and updating the code to have errors instead of seg faults.

For this change though, I haven't modified this code. In this PR I only moved the event processing previously located in the "OTA_EventProcessingTask" function into a helper function. This allows me to manipulate the state machine without entering the "while( otaAgent.state != OtaAgentStateStopped )" loop in the unit tests.

So this is an issue, but it's not introduced in this change so I'd like to merge this PR and then fix the lack of null checking in a separate PR if that's okay with you.